unit/instance.py

```
@@ -201,129 +233,75 @@ def set_path_m_time(path, m_time):
-                uuid = '{uuid}',
-                alias = '{alias}',
-                status = '{status}',
-                roles = {roles},
-                weight = {weight},
-                all_rw = {all_rw},
-                vshard_group = {vshard_group},
-                servers = {servers},
-            }})
-        '''.format(
-            uuid=r['uuid'],
-            alias=r['alias'],
-            status=r.get('status', 'healthy'),
-            roles='{{ {} }}'.format(', '.join([
-                '"{}"'.format(role) for role in r['roles']
-            ])),
-            weight=weight,
-            all_rw=all_rw,
-            vshard_group=vshard_group,
-            servers=r_servers
-        ))
-
-    def add_unjoined_server(self, alias, uri, status='healthy'):
-        self.eval('''
-            require('cartridge').internal.add_unjoined_server({{
-                uri = '{uri}',
-                alias = '{alias}',
-                status = '{status}',
-            }})
-        '''.format(
-            uri=uri,
-            alias=alias,
-            status=status,
-        ))
+        self.eval_res_err('''
+            local name, value = ...
+            require('cartridge').internal.set_variable(name, value)
+        ''', name, value)
+
+    def set_box_cfg(self, **new_box_cfg):
+        self.eval_res_err('''
+            require('cartridge').internal.set_box_cfg(...)
+        ''', new_box_cfg)
+
+    def add_membership_members(self, members):
+        self.eval_res_err('''
+            require('cartridge').internal.add_membership_members(...)
+        ''', members)
+
+    def add_replicaset(self, **kwargs):
+        self.eval_res_err('''
+            require('cartridge').internal.add_replicaset(...)
+        ''', kwargs)
+
+    def bootstrap_cluster(self):
+        res, err = self.eval_res_err('''
+            require('cartridge').internal.bootstrap_cluster()
+        ''')
+        assert err is None, err
```

What about to check errors in other places? Or maybe return an error?

---

I've added asserts in Lua code for internal functions. `eval_res_err` will raise exceptions for this errors with assert message.